### DISCUSSION TITLE
Publish routers on npm?

### DISCUSSION THREAD

**Discussion by icetbr** [honojs/hono#1481] [General]
Hi there, you have "the fastest router in the JavaScript world". Would you consider publishing it on npm? Have you though about monorepoing the project?

Thanks.

**Comment by yusukebe** [comment]
We don't have the plans to publish the routers on npm and make this "core" project as monorepo.

The routers are customized to Hono and we want to keep them in the project for maintenance without worrying about breaking changes.

The reason we don't use monorepo is because it is easier to maintain. Of course I know that a project with a good package manager like the middleware repo will make the maintenance easier, but we don't make the core project monorepo.

**Comment by icetbr** [comment]
That's understandable. I can of course use hono as a dependency in my project and make use of the router. Perhaps one day I will extract the router portion and, pending your blessing, maintain it as a isolated package.

Thank you for the response and your work on hono.
